### PR TITLE
Add config option "span_stack_trace_min_duration"

### LINKED ISSUE #1529: [META 94] Implement span_stack_trace_min_duration config option
Implement https://github.com/elastic/apm/issues/94

Spec PR: https://github.com/elastic/apm/pull/454/files

### PR DESCRIPTION
**Pull request by z1c0**
Fixes #1529

Introduce the new configuration option `span_stack_trace_min_duration` that replaces the now deprecated option `span_frames_min_duration`.
*Note*:
* The deprecated option will still be evaluated iff the new option is not present or has a default value set.
* The semantics of negative and 0 values have changed compared to the legacy setting (`span_frames_min_duration`).
  * Negative value will result in never capturing the stack traces.
  * A value of `0` (regardless of the time unit suffix) will always capture the stack traces.

The `Span` class was extended with the methods 
* `IsCaptureStackTraceOnStartEnabled`
* `IsCaptureStackTraceOnEndEnabled`
which evaluate the conditions for both versions of these settings which must be met so that stack traces are captures.
`SpanTests` exercise these new methods.

### CODE REVIEW

**Review by gregkalapos — COMMENTED** [review]
Nice, code looks ok to me! Added 2 small points to the code.

The docs build is broken - that should be fixed before we merge.

A few info on that one and how you can figure this out:
- You see 1 failing check at the bottom of the PR here in GitHub (elasticsearch-ci/docs — Build finished. - that one is red). 
- If you click on it, you end up in Jenkins - if you jump to `Console Output` in jenkins (or look at the logs on another UI) you'll see at the end that it fails, and it also prints: `08:02:22 INFO:build_docs:asciidoctor: WARNING: invalid reference: config-span-frames-min-duration` 
- `config-span-frames-min-duration` is a link in the doc which this PR removes - By searching for it: https://github.com/elastic/apm-agent-dotnet/search?q=config-span-frames-min-duration  I see the same link in `[docs/performance-tuning.asciidoc](https://github.com/elastic/apm-agent-dotnet/blob/20fea7888fd3a044e13fb34c3e657d26502a6911/docs/performance-tuning.asciidoc)`and in `[docs/troubleshooting.asciidoc](https://github.com/elastic/apm-agent-dotnet/blob/9ea92f0167744007ae2a0de5968fe8e28f6b5166/docs/troubleshooting.asciidoc)`  - we should update those to the new link name as well. Alternatively we can keep the old link name (but use the new config name in the rendered doc). I think renaming the link and updating it is a cleaner way to go.

**Comment by gregkalapos** [comment]
Adding @elastic/obs-docs for review: there is a minor doc change here.

File: src/Elastic.Apm/Model/Span.cs [hunk 1]
```diff
@@ -127,6 +129,43 @@ private Span(double duration, string id, string name, string parentId)
...
+		{
+			if (Configuration.StackTraceLimit != 0)
+			{
+				// If the legacy setting (span_frames_min_duration) is present but the new
+				// setting (span_stack_trace_min_duration) is not (or has a default value), the legacy setting dominates.
+				if (Configuration.SpanFramesMinDurationInMilliseconds != ConfigConsts.DefaultValues.SpanFramesMinDurationInMilliseconds &&
+				    Configuration.SpanStackTraceMinDurationInMilliseconds == ConfigConsts.DefaultValues.SpanStackTraceMinDurationInMilliseconds)
+				{
+					return Configuration.SpanFramesMinDurationInMilliseconds != 0;
+				}
```
[the remark this answers is not in the document]
Comment (z1c0) [reply src/Elastic.Apm/Model/Span.cs]: As suggested, I fixed the warnings (pragma and double comparison).
Additionally, I introduced `UseLegacyCaptureStackTraceSetting` for better readability and deduplication of the double comparison code.

**Review by gregkalapos — APPROVED** [review]
LGTM, thanks for the changes.

**Review by bmorelli25 — APPROVED** [review]
Docs LGTM

**Pull request merged**